polaris-core/src/main/java/org/apache/polaris/core/persistence/BasePersistence.java

```diff
@@ -37,6 +37,11 @@
  * defining the internal data model for Polaris, and which defines the basis for the RBAC model
  * provided by Polaris.
  *
+ * <p>Each method in this interface must be atomic, meaning that write operations must either fully
+ * succeed with all changes applied, or fail entirely without partial updates. Read operations must
+ * provide a consistent snapshot of the data as it existed at the start of the operation, ensuring
+ * snapshot isolation.
```

Comment:
"snapshot isolation" looks like an overloaded term to me... What do you mean by it exactly?

How about `Read operations must provide a consistent view of the data as it existed at the start of the operation`?

Reply:
I'm fine with the term, https://en.wikipedia.org/wiki/Snapshot_isolation, but I will update per your suggestion.

Reply:
The definition from Wikipedia does not hold given the reference to "at the start of the operation" earlier in the text... IMHO (note that `BasePersistence` does not define transactions).

Thanks for the update.